ff/serving/metadata/dashboard/dashboard_metadata.go

```diff
@@ -213,23 +389,34 @@ func (m *MetadataServer) readFromSource(source *metadata.Source) (map[string]Sou
 		return nil, fetchError
 	}
 	for _, variant := range variants {
-		variantMap[variant.Variant()] = SourceVariantResource{
-			Created:      variant.Created(),
-			Description:  variant.Description(),
-			Name:         variant.Name(),
-			Type:         variant.Type(),
-			Variant:      variant.Name(),
-			Owner:        variant.Owner(),
-			Provider:     variant.Provider(),
-			Labels:       variant.Labels(),
-			Features:     variant.Features(),
-			TrainingSets: variant.TrainingSets(),
+
+		sourceResource := sourceShallowMap(variant)
+		if singleResource {
+			//TODO make these channels so they don't wait for each other to finish to send their requests
```

Comment:
Since these three parts need to be separate calls, should I make them async and have a channel wait for them to finish?